### PR TITLE
UI support for publishing services

### PR DESCRIPTION
**Pull request by phantomjinx**
Please see commit messages and hopefully they should provide the necessary explanations for changes made.

This should be merged alongside [teiid-komodo-PR-37](https://github.com/teiid/teiid-komodo/pull/37)

### CODE REVIEW

File: src/main/ngapp/src/app/dataservices/shared/dataservice.service.ts
Comment (elvisisking) [inline src/main/ngapp/src/app/dataservices/shared/dataservice.service.ts:381]: Wdyt about moving this under core/utils?

File: src/main/ngapp/src/app/dataservices/shared/dataservice.service.ts
Comment (elvisisking) [inline src/main/ngapp/src/app/dataservices/shared/dataservice.service.ts:431]: Would be nice if this could also be implemented in `MockDataserviceService` so that the non-OpenShift dev mode would work.

File: src/main/ngapp/src/app/dataservices/shared/dataservice.service.ts
Comment (elvisisking) [inline src/main/ngapp/src/app/dataservices/shared/dataservice.service.ts:475]: Would be nice if this could also be implemented in `MockDataserviceService` so that the non-OpenShift dev mode would work.

**Review by elvisisking — APPROVED** [review]

**Review by mdrillin — APPROVED** [review]
Couple thing which need to be added
- Add published icon on Dataservice list view
- Add download action to Dataservice list view menu
I'm ok to merge and make these additions after merge.

I created TEIIDTOOLS-396 and -397 to address the above issues.

**Comment by phantomjinx** [comment]
Think all questions answered and follow-on issues logged so will merge this. If anything further is required then can rework in new PRs.

**Pull request merged**
